db/system_keyspace.cc

```diff
@@ -3032,11 +3030,9 @@ future<> system_keyspace::get_compaction_history(compaction_history_consumer&& f
             if (row.has("rows_merged")) {
                 entry.rows_merged = row.get_map<int32_t, int64_t>("rows_merged");
             }
-            return consumer(std::move(entry)).then([] {
-                return stop_iteration::no;
-            });
+            co_await consumer(std::move(entry));
+            co_return stop_iteration::no;
         });
```

Comment:
Idea for followup: while checking that `query_internal()` does keep the lambda alive (it seem to be safe) I noticed `query_processor::for_each_cql_result()` is ripe for coroutinization too.

Reply:
Kefu found them already https://github.com/scylladb/scylladb/pull/13621